config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/util/keystore/KeystoreHelper.java

```diff
@@ -0,0 +1,179 @@
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.*;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.util.Base64;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.security.KeyStore.getInstance;
+import static java.util.stream.StreamSupport.stream;
+
+@Singleton
+@SuppressWarnings("squid:S2068") // sonarcloud believes 'password' field names may have hardcoded passwords
+public class KeystoreHelper {
+
+    public static final String ENV_VAR_KEYSTORE_PATH = "ENV.KEYSTORE_PATH";
+
+    private static final String ENV_VAR_KEY_PASSWORD_FORMAT = "ENV.%s.PASSWORD";
+    private static final SecureRandom RANDOM = new SecureRandom();
+    private final char[] keystorePassword;
+
+    KeystoreHelper() {
+        // Generate a random passphrase with any type of char and using a secure random generator, in order to encrypt the secrets.
+        final String password = RandomStringUtils.random(64, 0, 0, true, true, null, RANDOM);
+        this.keystorePassword = Base64.getEncoder().encodeToString(password.getBytes(StandardCharsets.UTF_8)).toCharArray();
+    }
+
+    /**
+     * Create a PKCS12 KeyStore and write it to a byte array, to be used during the environment application.
+     *
+     * @param keystorePath path where the key files are located
+     * @param environmentProperties env properties with the key passwords
+     * @return a byte array containing the keystore data
+     */
+    public byte[] createKeyStoreFromEnvironment(String keystorePath, Map<String, String> environmentProperties) {
+        final Path keystoreDir = Paths.get(keystorePath);
+        final List<PrivateKey> privateKeys = stream(keystoreDir.spliterator(), false)
+                .filter(p -> p.endsWith(".p12"))
+                .map(p -> createPrivateKeyFromEnvironment(p, environmentProperties))
+                .filter(p -> p.getKeyPassword() != null) // ignoring keys with unspecified passwords
+                .collect(Collectors.toList());
+        return createKeyStore(privateKeys);
+    }
+
+    private PrivateKey createPrivateKeyFromEnvironment(Path privateKeyFile, Map<String, String> environmentProperties) {
+        PrivateKey privateKey = new PrivateKey();
+        privateKey.setAlias(FilenameUtils.getName(privateKeyFile.toFile().getName()));
+        privateKey.setPrivateKeyFile(() -> Files.newInputStream(privateKeyFile));
+        privateKey.setKeyPassword(environmentProperties.get(String.format(ENV_VAR_KEY_PASSWORD_FORMAT, privateKey.getAlias())));
+        return privateKey;
+    }
+
+    @VisibleForTesting
+    byte[] createKeyStore(Collection<PrivateKey> privateKeys) {
+        // if no private Keys specified do nothing
+        if (privateKeys.isEmpty()) {
+            return null;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Return an empty array instead of null. ([squid:S1168](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS1168))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=76&issues=AWZfeJbdDZywvPe4Lw3p&open=AWZfeJbdDZywvPe4Lw3p)

[ISSUE_KEY]: AWZfeJbdDZywvPe4Lw3p